app/src/main/java/com/portfolio/tasky/viewModels/LoginViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.portfolio.tasky.viewModels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.portfolio.tasky.usecases.EmailPatternValidator
+import com.portfolio.tasky.usecases.PasswordPatternValidator
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val emailValidator: EmailPatternValidator,
+    private val passwordPatternValidator: PasswordPatternValidator
+): ViewModel() {
+
+    private val mutableEmailChange = MutableLiveData(false)
+    val emailChange : LiveData<Boolean> = mutableEmailChange
+
+    private val mutablePasswordChange = MutableLiveData(false)
+    val passwordChange : LiveData<Boolean?> = mutablePasswordChange
```

Comment:
You can change Boolean? for Boolean

Reply:
Forgot this one as well ::face_palm::

Done